### PR TITLE
BUGFIX: Fixup #4679 constraints in node references

### PR DESCRIPTION
**Pull request by mhsdesign**
**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by mhsdesign** [comment]
It seems though that we need a test for this ^^

File: Neos.ContentRepository.Core/Classes/NodeType/ConstraintCheck.php [hunk 1]
```diff
@@ -14,6 +14,15 @@
     private function __construct(
         private array $constraints
     ) {
+        foreach ($this->constraints as $constraint => $value) {
```
Comment (kitsunet) [inline Neos.ContentRepository.Core/Classes/NodeType/ConstraintCheck.php:17]: Shouldn't we build a `NodeTypes` collection similar to `Projections` and use that as argument here?

Comment (mhsdesign) [reply Neos.ContentRepository.Core/Classes/NodeType/ConstraintCheck.php:17]: idk its your code ^^

Comment (mhsdesign) [reply Neos.ContentRepository.Core/Classes/NodeType/ConstraintCheck.php:17]: i would argue the `ConstraintCheck` is already this collection

Comment (kitsunet) [reply Neos.ContentRepository.Core/Classes/NodeType/ConstraintCheck.php:17]: Right, probably true, I guess I expected a "fromArray" or somthing.

**Review by kitsunet — APPROVED** [review]

**Pull request merged**